### PR TITLE
update docs for Oauth based key setup

### LINKED ISSUE #3053: Update Contribution Guide
I suggest moving contribution instructions to a standard `CONTRIBUTING.md` and updating the content from README.md. The current information about obtaining an api key still references the old chrome extension based build, and should be updated to include new instructions about creating your own test app and getting your oauth credentials and providing them to your development instance.

Requirements for PR:
- [ ] include docs on scopes needed/used by DIM
- [ ] include docs about using a publicly available IP for your development env callback url
- [ ] document where/how to provide your api credentials to app config in your dev env
- [ ] add link to discord server for collaboration questions https://discord.gg/UK2GWC7

*Note: I am planning to work on #3052 and will prepare a PR for this issue as I set up to start on that.*

### PR DESCRIPTION
**Pull request by LongLiveCHIEF**
- moved Developer Quick Start docs to standard `CONTRIBUTING.md` for better integration with github issues et. al.
- Updated API Key docs to reflect Oauth based auth key creation and setup

Fixes #3053

### CODE REVIEW

**Comment by bhollis** [comment]
Any reason to move `CONTRIBUTING.md` out of `docs/`? We moved it in there to avoid clutter at the root of the repo, and GitHub will use it from out of there: https://help.github.com/articles/setting-guidelines-for-repository-contributors/#adding-a-contributing-file

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+First, thank you for contributing to DIM! We're a community-driven project and we appreciate improvements large and small.
+
+Here are some tips to make sure your pull request can be merged smoothly:
+
+1. If you want to add a feature or make some change to DIM, consider [filing an issue](https://github.com/DestinyItemManager/DIM/issues/new) describing your idea first. This will give the DIM community a chance to discuss the idea, offer suggestions and pointers, and make sure what you're thinking of fits with the style and direction of DIM. If you want a more free-form chat, [join our Discord](https://discordapp.com/invite/UK2GWC7).
+2. Resist the temptation to change more than one thing in your PR. Keeping PRs focused on a single change makes them much easier to review and accept. If you want to change multiple things, or clean up/refactor the code, make a new branch and submit those changes as a separate PR.
+3. Please follow the existing style of DIM code when making your changes. While there's certainly room for improvement in the DIM codebase, if everybody used their favorite code style nothing would match up.
```
Comment (bhollis) [inline CONTRIBUTING.md]: This is probably obsolete now that we use Prettier.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Developer Quick start
+Clone the repo:
+
+* `git clone https://github.com/DestinyItemManager/DIM.git`
+
+Install dependencies:
+
+* Install [NodeJS](https://nodejs.org/).
+* Install [Yarn](https://yarnpkg.com/en/docs/install). If you're used to NPM, see "[Migrating from NPM](https://yarnpkg.com/lang/en/docs/migrating-from-npm/)". If you were already using NPM with DIM, run `yarn` to convert to Yarn. **Note (2018/4/25): Yarn from Homebrew will come with NodeJS 10, which isn't ready for prime time. Either install via `npm install -g yarn` or downgrade Node after installing.**
```
Comment (bhollis) [inline CONTRIBUTING.md]: Could you remove the NodeJS 10 warning? It's out of date.

Comment (LongLiveCHIEF) [reply CONTRIBUTING.md]: Sure thing!

File: CONTRIBUTING.md (hunk 1 above)
Comment (LongLiveCHIEF) [reply CONTRIBUTING.md]: Ok, I'll take it out.

**Comment by LongLiveCHIEF** [comment]
Sorry for not getting to these @bhollis. I've been super sick the last few days. I'll clean up the two things you mentioned and move `CONTRIBUTING.md` back to `/docs`.  Look for these in the next 24 hours.

I didn't realize github would load the integrated functionality for it from the `/docs` folder, and that was the only reason for the move.

**Comment by bhollis** [comment]
Thanks! Hope you're feeling better!

**Pull request merged**